syft/pkg/cataloger/binary/classifiers.go

```diff
@@ -78,7 +78,7 @@ func DefaultClassifiers() []Classifier {
 			FileGlob: "**/redis-server",
 			EvidenceMatcher: evidenceMatchers(
 				FileContentsVersionMatcher(`(?s)payload %5.*?(?P<version>\d.\d\.\d\d*)[a-z0-9]{12,15}-[0-9]{19}`),
-				FileContentsVersionMatcher(`(?s)\x00(?P<version>\d.\d\.\d\d*)[a-z0-9]{12}-[0-9]{19}\x00.*?payload %5`),
+				FileContentsVersionMatcher(`(?s)\x00(?P<version>\d.\d\.\d\d*)[a-z0-9]{12,15}-[0-9]{19}\x00.*?payload %5`),
```

Comment:
This regex can match a large segment of the file (large enough that a snippet can't be added to the PR and still pass the "no big binaries" lint).

I spent a little time trying to work out regex that still performed as well at matching redis server binaries but didn't have this drawback, and gave up for not having enough time. But it would be a nice follow up if someone can figure out how.

Discussed with @wagoodman offline and decided to approve as-is, since I couldn't make the regex shorter within the time box.